tools/serverpod_cli/lib/src/analyzer/models/definitions.dart

```diff
@@ -98,6 +97,33 @@ class SerializableModelFieldDefinition {
 
   final bool shouldPersist;
 
+  // default value
+  final dynamic defaultVal;
+
+  // default model value
+  final dynamic defaultModelVal;
+
+  // default valdatabase
+  final dynamic defaultDatabaseVal;
+
+  /// returns field default value for models
+  dynamic get modelDefaultValue => defaultModelVal ?? defaultVal;
+
+  /// returns field default value for database
+  dynamic get databaseDefaultValue => defaultDatabaseVal ?? defaultVal;
+
+  /// returns true if one of the defauls its not null
+  bool get hasDefauls =>
+      defaultVal != null ||
+      defaultModelVal != null ||
+      defaultDatabaseVal != null;
+
+  /// returns true if only has database default
+  bool get hasOnlyDatabaseDefauls =>
+      defaultVal == null &&
+      defaultModelVal == null &&
+      defaultDatabaseVal != null;
```

Comment:
"hasOnlyDatabaseDefauls" is used to check if the field has only database default and then we check if the field is nullable or not , because we dont allow non nullable fields for this case, check "restrictions.dart" file.